src/Plugin/AddPathPlugin.php

```diff
@@ -48,7 +48,7 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
     {
         $identifier = spl_object_hash((object) $first);
 
-        if (!array_key_exists($identifier, $this->alteredRequests)) {
+        if (!array_key_exists($identifier, $this->alteredRequests) || 0 !== strpos($request->getUri()->getPath(), $this->uri->getPath())) {
```

Comment:
should we assign $request->getUri()->getPath() and $this->uri->getPath() to variables to not execute the code twice, in the expected case of having to add the prefix?